src/react/utils.js

```diff
@@ -850,3 +850,71 @@ export function sanitizeReactElementForFirstRenderOnly(realm: Realm, reactElemen
   }
   return reactElement;
 }
+
+// _forEachBindingOfEffects(effects: Effects, func: (binding: Binding) => void): void {
+//   let [result, , nestedBindingsToIgnore] = effects.data;
+//   for (let [binding] of nestedBindingsToIgnore) {
+//     func(binding);
+//   }
+//   if (result instanceof PossiblyNormalCompletion || result instanceof JoinedAbruptCompletions) {
+//     this._forEachBindingOfEffects(result.alternateEffects, func);
+//     this._forEachBindingOfEffects(result.consequentEffects, func);
+//   }
+// }
+
+// _hasWriteConflictsFromReactRenders(
+//   bindings: Set<Binding>,
+//   effects: Effects,
+//   nestedEffectsList: Array<Effects>,
+//   evaluatedNode: ReactEvaluatedNode
+// ): boolean {
+//   let ignoreBindings = new Set();
+//   let failed = false;
+//   // TODO: should we also check realm.savedEffects?
+//   // ref: https://github.com/facebook/prepack/pull/1742
+
+//   for (let nestedEffects of nestedEffectsList) {
+//     this._forEachBindingOfEffects(nestedEffects, binding => {
+//       ignoreBindings.add(binding);
+//     });
+//   }
+
+//   this._forEachBindingOfEffects(effects, binding => {
+//     if (bindings.has(binding) && !ignoreBindings.has(binding)) {
+//       failed = true;
+//     }
+//     bindings.add(binding);
+//   });
+
+//   if (failed) {
+//     evaluatedNode.status = "WRITE-CONFLICTS";
+//   }
+//   return failed;
+// }
+
+function checkForWriteBindings(effects: Effects): void {
+  let [result, , writeBindings] = effects.data;
+  for (let [binding] of writeBindings) {
+    debugger;
```

Comment:
Intentionally left in so you can see it correctly finds an impure binding modification.